File: app/src/main/java/com/example/exerciseinterview/SelectionSort.java
package com.example.exerciseinterview;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arrInput = {64, 23, 63, 2, 3, 0};
        selectionSort(arrInput);
        System.out.println("Sorted array: " + arrayToString(arrInput));
    }

    public static void selectionSort(int[] arr) {
        int arrSize = arr.length;

        for (int i = 0; i < arrSize; i++) {
            int currentMinIndex = i;

            for (int j = i + 1; j < arrSize; j++) {
                if (arr[j] < arr[currentMinIndex]) {
                    currentMinIndex = j;
                }
            }

            // Swap elements
            int temp = arr[currentMinIndex];
            arr[currentMinIndex] = arr[i];
            arr[i] = temp;
        }
    }

    public static String arrayToString(int[] arr) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < arr.length; i++) {
            sb.append(arr[i]);
            if (i < arr.length - 1) {
                sb.append(", ");
            }
        }
        return sb.toString();
    }
}

File: app/src/main/java/com/example/exerciseinterview/SelectionSort.kt
package com.example.exerciseinterview

//Selection Sort : https://www.geeksforgeeks.org/selection-sort/
//Comparing value one-by-one, searching for the smallest/largest value, swap it from unsorted portiion to sorted portion

fun main(args: Array<String>) {

    var arrInput = intArrayOf(64,23,63,2,3,0)
    selectionSort(arrInput)

    println("Sorted array: ${arrInput.joinToString()}")

}

fun selectionSort (arr: IntArray){

    var arrSize = arr.size

    //outer loops iterate through array
    for (i in 0..<arrSize){
        //set min. value always start with 0 (unsorted array)
        var currentMint = i
        //inner loops to compare the value & swapping
        for (j in i+1 until arrSize){
            if (arr[j] < arr[currentMint] ){
                //if arr j less than current min, set the current min to j index
                currentMint = j
            }
        }

        //do swapping

        var temp = arr[currentMint]
        arr[currentMint] = arr[i]
        arr[i] = temp
    }

}